cloudimg/aws.py

```diff
@@ -413,14 +416,27 @@ def upload_to_container(self, image_path, container_name, object_name,
         if image_path.lower().startswith('http'):
             # Stream the upload from a remote URL
             log.info('Opening stream to: %s', image_path)
-            resp = requests.get(image_path, stream=True)
+            resp = requests.get(image_path, stream=True, timeout=30)
             resp.raise_for_status()
             stream = resp.iter_content(chunk_size)
             callback = UploadProgress(container_name, object_name)
+            if object_name.endswith(".xz"):
+                raise NotImplementedError
```

Comment:
I think having a message like "LZMA decompression is not implemented for S3 content from an HTTP source" would be useful.

People do sometimes throw away / fail to log exception backtraces, so it's useful for there to be some greppable message which can reliably point back at the code raising the exception.